net/net.c

```diff
@@ -134,6 +149,74 @@ M0_INTERNAL bool m0_net_endpoint_is_valid(const char *endpoint)
 }
 #endif /* !__KERNEL__ */
 
+void m0_net_xprt_default_set(struct m0_net_xprt *xprt)
+{
+	M0_ENTRY();
+	M0_LOG(M0_DEBUG, "setting default xprt to %p:%s", xprt, xprt->nx_name);
+	xprt_default = xprt;
+	m0_net_xprt_register(xprt);
+}
+M0_EXPORTED(m0_net_xprt_default_set);
+
+struct m0_net_xprt *m0_net_xprt_default_get(void)
+{
+	M0_ENTRY();
+	M0_LOG(M0_DEBUG, "getting default xprt to %p:%s",
+			  xprt_default,
+			  xprt_default?xprt_default->nx_name:"NULL");
+	return xprt_default;
+}
+M0_EXPORTED(m0_net_xprt_default_get);
+
+struct m0_net_xprt **m0_net_all_xprt_get(void)
+{
+	M0_ENTRY();
+	return xprts;
+}
+M0_EXPORTED(m0_net_all_xprt_get);
+
+int m0_net_xprt_nr_get(void)
```

Comment:
Maybe we should not allow xprt de-register?

Otherwise this will imply a lot of problems:
* if a xprt is in use, and it is de-regiestered, this will definitely introduce problems.
* So, we need to have refcount for all xprt?

Reply:
incorporated